### PR TITLE
Since notify might not wake up the right thread, notifyAll should be used instead.

### PR DESCRIPTION
**Pull request by kamaci**

### CODE REVIEW

File: extensions-contrib/druid-rocketmq/src/main/java/org/apache/druid/firehose/rocketmq/RocketMQFirehoseFactory.java [hunk 1]
```diff
@@ -404,7 +404,7 @@ public void putRequest(final DruidPullRequest request)
         this.requestsWrite.add(request);
         if (!hasNotified) {
           hasNotified = true;
-          notify();
+          notifyAll();
```
Comment (gianm) [inline extensions-contrib/druid-rocketmq/src/main/java/org/apache/druid/firehose/rocketmq/RocketMQFirehoseFactory.java]: The class that this extends uses this pattern extensively:

```java
        synchronized (this) {
            if (!this.hasNotified) {
                this.hasNotified = true;
                this.notify();
            }
        }
```

I don't think we need to second-guess it. Maybe revert back to `notify` and add a comment about why `notifyAll` is not used?

Comment (kamaci) [reply extensions-contrib/druid-rocketmq/src/main/java/org/apache/druid/firehose/rocketmq/RocketMQFirehoseFactory.java]: 👍

File: extensions-contrib/druid-rocketmq/src/main/java/org/apache/druid/firehose/rocketmq/RocketMQFirehoseFactory.java [hunk 2]
```diff
@@ -404,6 +404,7 @@ public void putRequest(final DruidPullRequest request)
         this.requestsWrite.add(request);
         if (!hasNotified) {
           hasNotified = true;
+          // No need to use notifyAll here since extended class com.alibaba.rocketmq.common.ServiceThread handles it
```
Comment (gianm) [inline extensions-contrib/druid-rocketmq/src/main/java/org/apache/druid/firehose/rocketmq/RocketMQFirehoseFactory.java]: Probably better:

> No need to use notifyAll here; following convention of parent class com.alibaba.rocketmq.common.ServiceThread

Comment (kamaci) [reply extensions-contrib/druid-rocketmq/src/main/java/org/apache/druid/firehose/rocketmq/RocketMQFirehoseFactory.java]: I've updated my PR 👍

Comment (kamaci) [reply extensions-contrib/druid-rocketmq/src/main/java/org/apache/druid/firehose/rocketmq/RocketMQFirehoseFactory.java]: @gianm is current changes OK for this PR?

**Comment by gianm** [comment]
Yes, it looks good. Thanks!

**Pull request merged**